Content.Shared/Nyanotrasen/Item/PseudoItem/SharedPseudoItemSystem.Checks.cs

```diff
@@ -0,0 +1,165 @@
+﻿using Content.Shared.Item;
+using Content.Shared.Storage;
+
+namespace Content.Shared.Nyanotrasen.Item.PseudoItem;
+
+/// <summary>
+/// Almost all of this is code taken from other systems, but adapted to use PseudoItem.
+/// I couldn't use the original functions because the resolve would fuck shit up, even if I passed a constructed itemcomp
```

Comment:
creating a new component to pass into :
1. would fail resolve since its not actually added to the entity
2. extremely sus

im guessing this is just to make it so a felenid cant be picked up instantly, would be better to just have that be an event raised for ItemComponent that felenids cancel, so they still reuse all the logic but cant just left click to pick up like a mouse

Reply:
Pseudoitems lack the itemcomponent when they're not inside a container because it brings a ton of unwanted features along with it. Even just adding it to run the check could result in weird shit happening if someone knew enough. This approach works "fine" for what its supposed to do

Reply:
The entire reason I cant use the original functions as is is because they expect the item to have an itemcomponent, and theres no way to tell it to use anything else, even if it doesn't directly do anything with said component